operatorcert/utils.py

```diff
@@ -0,0 +1,168 @@
+    """
+    Gets all the annotations from the bundle metadata
+
+    Args:
+        bundle_path (Path): A path to the bundle version
+
+    Returns:
+        A dict of all the annotation keys and values
+    """
+    paths = [
+        ("metadata", "annotations.yaml"),
+        ("metadata", "annotations.yml"),
+    ]
+    annotations_path = find_file(bundle_path, paths)
+    if not annotations_path:
+        raise RuntimeError("Annotations file not found")
+
+    with annotations_path.open() as fh:
+        content = yaml.safe_load(fh)
+        return content.get("annotations", {})
+
+
+def get_csv_annotations(bundle_path: pathlib.Path, package: str) -> Dict:
+    """
+    Gets all the annotations from the bundle CSV
+
+    Args:
+        bundle_path (Path): A path to the bundle version
+        package (str): Operator package name
+
+    Returns:
+        A dict of all the annotation keys and values
+    """
+    paths = [
+        ("manifests", f"{package}.clusterserviceversion.yaml"),
+        ("manifests", f"{package}.clusterserviceversion.yml"),
+    ]
+    csv_path = find_file(bundle_path, paths)
+    if not csv_path:
+        raise RuntimeError("Cluster service version (CSV) file not found")
+
+    with csv_path.open() as fh:
+        content = yaml.safe_load(fh)
+        return content.get("metadata", {}).get("annotations", {})
+
+
+def get_supported_indices(
+    pyxis_url: str, ocp_versions_range: str, max_ocp_version: str = None
+) -> List[str]:
+    """
+    Gets all the known supported OCP indices for this bundle.
+
+    Args:
+        pyxis_url (str): Base URL to Pyxis
+        ocp_versions_range (str): OpenShift version annotation
+        max_ocp_version (str): OLM property in the bundle CSV
+
+    Returns:
+        A list of supported OCP versions in descending order
+    """
+    url = urljoin(pyxis_url, "v1/operators/indices")
+
+    filter_ = f"organization==certified-operators"
+    if max_ocp_version:
+        filter_ += f";ocp_version=le={max_ocp_version}"
+
+    # Ignoring pagination. 500 OCP versions would be a lot for a single
+    # version of an operator to support.
+    params = {
+        "filter": filter_,
+        "ocp_versions_range": ocp_versions_range,
+        "page_size": 500,
+        "sort_by": "ocp_version[desc]",
+        "include": "data.ocp_version,data.path",
+    }
+
+    rsp = requests.get(url, params=params)
+    rsp.raise_for_status()
+    return rsp.json()["data"]
+
+
+def ocp_version_info(bundle_path: pathlib.Path, pyxis_url: str) -> Dict:
+    """
+    Gathers some information pertaining to the OpenShift versions defined in the
+    Operator bundle.
+
+    Args:
+        args: Parsed args from ArgumentParser
+
+    Returns:
+        A dict of pertinent OCP version information
+    """
+    bundle_annotations = get_bundle_annotations(bundle_path)
+
+    ocp_versions_range = str(bundle_annotations.get(OCP_VERSIONS_ANNOTATION))
+    if not ocp_versions_range:
```

Comment:
This won't work as expected I guess.. Because you use `.get()` with no default and then convert it to string the `ocp_versions_range` will contain a string `None` which is a valid string and this condition doesn't catch that.

Why do you convert the version to string? I think if you remove the conversion it should work as expected.

Reply:
Good catch, this wasn't intentional and was actually caught with the unit tests I was missing. It's fixed now.